apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/sampling/SamplingService.java

```diff
@@ -114,4 +113,26 @@ public void forceSampled() {
     private void resetSamplingFactor() {
         samplingFactorHolder = new AtomicInteger(0);
     }
+
+    /**
+     * Reset scheduledFuture to support samplingRate changed.
+     */
+    void resetScheduledFuture() {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
+        }
+        on = false;
+        if (samplingRateWatcher.getSamplingRate() > 0) {
+            on = true;
+            this.resetSamplingFactor();
+            ScheduledExecutorService service = Executors.newSingleThreadScheduledExecutor(
+                new DefaultNamedThreadFactory("SamplingService"));
+            scheduledFuture = service.scheduleAtFixedRate(new RunnableWithExceptionProtection(
+                this::resetSamplingFactor, t -> LOGGER.error("unexpected exception.", t)), 0, 3, TimeUnit.SECONDS);
+            LOGGER.debug(
+                "Agent sampling mechanism started. Sample {} traces in 3 seconds.",
+                samplingRateWatcher.getSamplingRate()
+            );
+        }
```

Comment:
I am confused about this, why need re-scheduling?

Reply:
`on=true/false` could be determined by `onChange` event.

Reply:
done.